src/FSharpPlus/Extensions/Array.fs

```diff
@@ -36,22 +37,89 @@ module Array =
     /// <returns>Array with values returned from mapping function.</returns>
     let lift3 mapping list1 list2 list3 =
         let lenx, leny, lenz = Array.length list1, Array.length list2, Array.length list3
-        let combinedFirstTwo = Array.init (lenx * leny) (fun i -> (list1.[i / leny], list2.[i % leny]))
+        let combinedFirstTwo = Array.init (lenx * leny) (fun i -> let (d, r) = Math.DivRem (i, leny) in (list1.[d], list2.[r]))
 
-        Array.init (lenx * leny * lenz) (fun i -> combinedFirstTwo.[i/leny], list3.[i%leny])
+        Array.init (lenx * leny * lenz) (fun i -> let (d, r) = Math.DivRem (i, leny) in combinedFirstTwo.[d], list3.[r])
         |> Array.map (fun x -> mapping (fst (fst x)) (snd (fst x)) (snd x))
 
     /// Concatenates all elements, using the specified separator between each element.
-    let intercalate (separator: _ []) (source: seq<_ []>) = source |> Seq.intercalate separator |> Seq.toArray
+    let intercalate (separator: 'T []) (source: seq<'T []>) =
+    #if FABLE_COMPILER || NET45
+        source |> Seq.intercalate separator |> Seq.toArray
+    #else
+        let mutable coll = new ArrayCollector<'T> ()
+        let mutable notFirst = false
+        source |> Seq.iter (fun element ->
+            if notFirst then coll.AddMany separator
+            coll.AddMany element
+            notFirst <- true)
+        coll.Close ()
+    #endif
 
     /// Inserts a separator element between each element in the source array.
-    let intersperse element source = source |> Array.toSeq |> Seq.intersperse element |> Seq.toArray : 'T []
+    let intersperse element (source: 'T []) =
+        match source with
+        | [||] -> [||]
+        | _ ->
+            let finalLength = Array.length source * 2 - 1
+            Array.init finalLength (fun i ->
+                match Math.DivRem (i, 2) with
+                | i, 0 -> source.[i]
+                | _    ->  element)
 
     /// Creates a sequence of arrays by splitting the source array on any of the given separators.
     let split (separators: seq<_ []>) (source: _ []) = source |> Array.toSeq |> Seq.split separators |> Seq.map Seq.toArray
 
     /// Replaces a subsequence of the source array with the given replacement array.
-    let replace oldValue newValue source = source |> Array.toSeq |> Seq.replace oldValue newValue |> Seq.toArray                : 'T []
+    let replace (oldValue: 'T seq) (newValue: 'T seq) (source: 'T[]) : 'T[] =
+    #if FABLE_COMPILER || NET45
+        source |> Array.toSeq |> Seq.replace oldValue newValue |> Seq.toArray: 'T []
+    #else
+        let oldValueArray = oldValue |> Seq.toArray
```

Comment:
I had to change to toArray here since the actual type signature in the released version of F#+ is seq as can be seen for instance here:
https://www.fuget.org/packages/FSharpPlus/1.2.5/lib/netstandard2.0/FSharpPlus.dll/FSharpPlus/Array
@fcallejon